mdn/src/main/java/org/apache/james/mdn/parsing/DispositionParser.java

```diff
@@ -0,0 +1,156 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mdn.parsing;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.james.mdn.action.mode.DispositionActionMode;
+import org.apache.james.mdn.fields.Disposition;
+import org.apache.james.mdn.fields.Field;
+import org.apache.james.mdn.modifier.DispositionModifier;
+import org.apache.james.mdn.sending.mode.DispositionSendingMode;
+import org.apache.james.mdn.type.DispositionType;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+
+public class DispositionParser implements FieldsParser.FieldParser {
+    private final boolean strict;
+
+    public DispositionParser(boolean strict) {
+        this.strict = strict;
+    }
+
+    @Override
+    public Optional<Field> parse(String value) {
+        Preconditions.checkNotNull(value);
+
+        int slashPosition = value.indexOf('/');
+        Optional<DispositionActionMode> actionMode = parseActionMode(value, slashPosition);
+        String afterActionMode = value.substring(slashPosition + 1);
+
+        int semiColonPosition = afterActionMode.indexOf(';');
+        Optional<DispositionSendingMode> sendingMode = parseSendingMode(afterActionMode, semiColonPosition);
+        String afterSendingMode = afterActionMode.substring(semiColonPosition + 1);
```

Comment:
you're not consistent about where you split the string : once in submethod, once in the body